src/main/java/org/jenkinsci/plugins/workinghours/WorkingHoursQueueTaskDispatcher.java

```diff
@@ -112,20 +114,20 @@ public boolean isReleased(EnforceBuildScheduleAction action,
     /**
      * Determines whether a queue item can run at the current moment.
      * @param itemActionable The item being checked.
-     * @param item The queue item to check.     
+     * @param item The queue item to check.
      * @return true if the item can run now; false otherwise.
      */
     public boolean canRunNow(Actionable itemActionable,
             Queue.Item item) {
-        Calendar now = Calendar.getInstance();
+        LocalDateTime now = LocalDateTime.now();
 
         EnforceBuildScheduleAction action = itemActionable.getAction(EnforceBuildScheduleAction.class);
 
         WorkingHoursPlugin config = ExtensionList.lookup(WorkingHoursPlugin.class).get(0);
 
         // Check whether today should be excluded according to the excluded dates we set.
         for (ExcludedDate excludedDate : config.getExcludedDates()) {
-            if(excludedDate.shouldExclude(now)){
+            if(excludedDate.shouldExclude(null)){
```

Comment:
This probably isn't a huge deal, but with this change, every call to `shouldExclude` will use a slightly different time, since it's generated in there. Would it be better to continue generating the date in this function?

Reply:
But it may be used in Tests to set a different day. And I'm confused whether we need to add useless code to fulfill tests?